src/OpenLoco/Message.cpp

```diff
@@ -0,0 +1,165 @@
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::vehicle, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::company, MessageItemArgumentType::null, MessageItemArgumentType::null },
+        { MessageItemArgumentType::vehicle, MessageItemArgumentType::company, MessageItemArgumentType::null },
+        { MessageItemArgumentType::vehicle, MessageItemArgumentType::company, MessageItemArgumentType::null },
+    };
+
+    const MessageItemType& getMessageItemArgumentType(const MessageType type)
+    {
+        return _MessageItemArgumentType[enumValue(type)];
+    }
+
+    // 0x004F8C22
+    static constexpr MessageCriticality messageCriticalities[31] = {
+        MessageCriticality::advice,
+        MessageCriticality::general,
+        MessageCriticality::advice,
+        MessageCriticality::minorCompany,
+        MessageCriticality::minorCompany,
+        MessageCriticality::minorCompany,
+        MessageCriticality::minorCompany,
+        MessageCriticality::minorCompany,
+        MessageCriticality::minorCompany,
+        MessageCriticality::advice,
+        MessageCriticality::advice,
+        MessageCriticality::majorCompany,
+        MessageCriticality::advice,
+        MessageCriticality::minorCompany,
+        MessageCriticality::minorCompany,
+        MessageCriticality::general,
+        MessageCriticality::majorCompany,
+        MessageCriticality::general,
+        MessageCriticality::general,
+        MessageCriticality::general,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompetitor,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompany,
+        MessageCriticality::majorCompetitor,
+    };
+
+    MessageCriticality getMessageCriticality(const MessageType type)
+    {
+        return messageCriticalities[enumValue(type)];
+    }
+
+    // 0x004F8C41
+    constexpr static Audio::SoundId messageSounds[31] = {
```

Comment:
I think we should consider introducing a struct that merges these four tables (messageCriticalities, messageSounds, _MessageTypeFlagsArr, _MessageItemArgumentType).

Reply:
already ahead of you :P